### PR TITLE
Add blog.

### PR DESCRIPTION
**Pull request by oluoluoxenfree**

### CODE REVIEW

File: old_source/blog/001-how-extend-works.html.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 title: How @extend Works
 author: Natalie Weizenbaum
-date: 2013-11-22 16:57 PST
+# date: 2013-11-22 16:57 PST
```
Comment (jgerigmeyer) [inline old_source/blog/001-how-extend-works.html.md]: Probably revert these changes -- I think ideally we don't need to make any changes directly to this `old_source` directory.

File: source/_layouts/blog.liquid [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+layout: base
+---
+
+{{ content }}
```
Comment (jgerigmeyer) [inline source/_layouts/blog.liquid]: This is probably where the attribution stuff can go.

File: source/blog.liquid [hunk 3]
```diff
@@ -0,0 +1,32 @@
+---
+pagination:
+  data: collections.blog
+  size: 5
+  alias: posts
+  reverse: true
+layout: blog
```
Comment (jgerigmeyer) [inline source/blog.liquid]:
```suggestion
layout: has_no_sidebars
```

File: source/blog/001-how-extend-works.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
+---
+title: How @extend Works
+author: Natalie Weizenbaum
+tags: blog
+layout: base
```
Comment (jgerigmeyer) [inline source/blog/001-how-extend-works.md]: We can move these shared pieces of data (plus `permalink`) into a `source/blog/blog.11tydata.yml` file.

**Review by jgerigmeyer — COMMENTED** [review]
👏

File: source/blog/001-how-extend-works.md (hunk 4 above)
Comment (jgerigmeyer) [inline source/blog/001-how-extend-works.md]:
```suggestion
layout: blog
```

**Comment by stacyk** [comment]
[port the blog](https://trello.com/c/J9iKTXHp/28-port-the-blog)

File: source/blog.liquid [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    <li>
+      <a href="{{ pagination.href.previous }}">Previous</a>
+    </li>
+  {% endif %}
+  {%- for post in posts -%}
+    <li>
+      <h2>
+        <a href="{{post.url}}">{{ post.data.title }}</a>
+      </h2>
+      <p>Posted {{ post.date | formatBlogDate }} by {{ post.data.author }}</p>
```
Comment (stacyk) [inline source/blog.liquid]: I believe we are missing this class:
```suggestion
      <p class="sl-c-attribution">Posted {{ post.date | formatBlogDate }} by {{ post.data.author }}</p>
```

File: source/blog.liquid [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      <p>{{ post.content | truncatePost }}</p>
+    </li>
+  {%- endfor -%}
+  {% if pagination.href.next %}
+    <li>
+      <a href="{{ pagination.href.next }}">Next</a>
```
Comment (stacyk) [inline source/blog.liquid]: The live site has "Next Page". Not sure if we wanted to stick with that or if we made this change intentionally.

File: source/blog.liquid [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      <a href="{{ pagination.href.previous }}">Previous</a>
+    </li>
+  {% endif %}
+  {%- for post in posts -%}
+    <li>
+      <h2>
+        <a href="{{post.url}}">{{ post.data.title }}</a>
+      </h2>
+      <p>Posted {{ post.date | formatBlogDate }} by {{ post.data.author }}</p>
+      <p>{{ post.content | truncatePost }}</p>
```
Comment (stacyk) [inline source/blog.liquid]: Does including this `<p>` tag cause an empty `<p></p>`?

File: source/blog.liquid (hunk 6 above)
Comment (stacyk) [reply source/blog.liquid]: I noticed that the IDs are no longer included on the headings (that aren't the post title itself). I am not positive, but I think we spoke about trying to get the same IDs on the headings that the live site has in order to not break a link that may be pointed to the site/#anchor. I don't know if that would apply here, but I think we were discussing it in regards to the work Sana is doing on the Learn Sass page.

**Review by stacyk — COMMENTED** [review]
@oluoluoxenfree I left a few notes below and I am including what I posted in Slack so it is all in one place:
1. the bullet points are due to the fact that we are now nesting the articles in a list (which is good, either a list or the <article> tag would be good)
2. There is a white bg that is missing on the pages, but that is fixed in the color-config brand so please ignore that

File: source/blog.liquid (hunk 7 above)
Comment (stacyk) [reply source/blog.liquid]: Screenshot 2023-03-07 at 1 48 35 PM

File: source/blog.liquid (hunk 6 above)
Comment (oluoluoxenfree) [reply source/blog.liquid]: Which IDs? If you mean the page anchors we moved that to separate ticket @stacyk

File: source/blog.liquid (hunk 7 above)
Comment (oluoluoxenfree) [reply source/blog.liquid]: I think it was yes, good spot!

File: source/blog.liquid (hunk 6 above)
Comment (oluoluoxenfree) [reply source/blog.liquid]: Oops, missed this one!

Comment (jgerigmeyer) [reply source/blog.liquid]: @oluoluoxenfree I'm in this code doing some review, so I can add this in 👍

Comment (jgerigmeyer) [reply source/blog.liquid]: https://trello.com/c/T0UW4E0y

**Review by jgerigmeyer — APPROVED** [review]
Looks good! @stacyk I think all issues are resolved and this is ready for final review.

File: source/blog.liquid [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  {% if pagination.href.previous %}
+    <p>
+      <a href="{{ pagination.href.previous }}">Previous page</a>
+    </p>
+  {% endif %}
+{% endif %}
+
+<ul class="sl-c-list">
+  {% for post in posts %}
+    <li>
```
Comment (jgerigmeyer) [inline source/blog.liquid:26]: I think the only difference in terms of markup from the old site to the new one is that we added a `<ul>` with `<li>`s instead of just a long list of `<p>` tags.

**Review by mirisuzanne — APPROVED** [review]
It all looks great to me!
